firebase-firestore/src/main/java/com/google/firebase/firestore/local/LruGarbageCollector.java

```diff
@@ -91,4 +157,79 @@ int removeTargets(long upperBound, SparseArray<?> activeTargetIds) {
   int removeOrphanedDocuments(long upperBound) {
     return delegate.removeOrphanedDocuments(upperBound);
   }
+
+  Results collect(SparseArray<?> activeTargetIds) {
+    if (params.minBytesThreshold == Params.COLLECTION_DISABLED) {
+      Logger.debug("LruGarbageCollector", "Garbage collection skipped; disabled");
+      return Results.DidNotRun();
+    }
+
+    long cacheSize = getByteSize();
+    if (cacheSize < params.minBytesThreshold) {
+      Logger.debug(
+          "LruGarbageCollector",
+          "Garbage collection skipped; Cache size "
+              + cacheSize
+              + " is lower than threshold "
+              + params.minBytesThreshold);
+      return Results.DidNotRun();
+    } else {
+      return runGarbageCollection(activeTargetIds);
+    }
+  }
+
+  private Results runGarbageCollection(SparseArray<?> liveTargetIds) {
+    long start = System.currentTimeMillis();
+    int sequenceNumbers = calculateQueryCount(params.percentileToCollect);
+    // Cap at the configured max
+    if (sequenceNumbers > params.maximumSequenceNumbersToCollect) {
+      Logger.debug(
+          "LruGarbageCollector",
+          "Capping sequence numbers to collect down to the maximum of "
+              + params.maximumSequenceNumbersToCollect
+              + " from "
+              + sequenceNumbers);
+      sequenceNumbers = params.maximumSequenceNumbersToCollect;
+    }
+    long countedTargets = System.currentTimeMillis();
```

Comment:
Can you someone indicate that these are timestamps? This confused me at first, as I was expecting this variable to contain the count of targets.